scripts/check_compliance.py

```diff
@@ -269,12 +269,12 @@ def check_no_undef_within_kconfig(self, kconf):
         Checks that there are no references to undefined Kconfig symbols within
         the Kconfig files
         """
-        undef_ref_warnings = [warning for warning in kconf.warnings
-                              if "undefined symbol" in warning]
+        undef_ref_warnings = "\n\n\n".join(warning for warning in kconf.warnings
+                                           if "undefined symbol" in warning)
```

Comment:
I miss the benefit(s) of this change

Reply:
Gets rid of a temporary list, though I might be the only one who cares. :P

Reply:
A list is a "higher level" / more structured type than the same list flattened to a string. That's the part that annoys me a bit but I might be the only one who cares :-)

Reply:
Yeah, I get what you're saying. Could still read it as "if there's any warning text" to have it make sense at least. :)